### ISSUE TITLE
Light up non-octave notes in keyboard mode

### ISSUE THREAD

**Issue by faroit** [jrcurtis/subsequencely#7]
For me the grid keyboard is hard to play as a keyboard player because it misses the visual cues from the black and white keys. 

The stock launchpad firmware lights up the white keys. That way it is easier to visually memorise at which position in the current octave I am. Therefore I would suggest to include this visual help as well.

Since you already support user defined scales it could be implemented in a way where you light up the black and white keys in two different colours (I would suggest to use lower brightness than the octave pads) as shown in the grid setup mode.

**Comment by jrcurtis** [comment]
That's very possible. I was a little hesitant to add this before because I didn't want the view to become too busy and I wasn't sure what kind of highlighting scheme would work best. I'll give your suggestion a try and see how it works out.

**Comment by faroit** [comment]
okay I agree. Maybe also interesting: http://fabriziopoce.com/isocontrollers.html They have a lot of more nice keyboard views implemented. Some of them might be worth checking out

**Comment by jrcurtis** [comment]
Yeah! The demo for that was one of the first videos I saw of the Launchpad Pro. Really neat program.

I did some work on the colors, but I'm not sure how much it's helped. The way I've got it now, it lights up the root note the same as before, the white key notes in blue, and lights C in green. I left the black key notes unlit because I thought it was too many colors at once, but the code is there to light them and you can customize all the colors in colors.c.

This scheme works well enough in the default chromatic layout, but in any other scale it becomes less useful since the white keys are no longer in recognizable patterns, and C can't be used as a reference point if it isn't in the scale at all.

The ISOControllers layouts light notes up by scale degree, which helps them feel a little more consistent, but I was messing around with them tonight and I had trouble with how the same note can be different colors in different places on the grid in most of the layouts. Might be worth trying something like that, though. I'm open to suggestions.

**Comment by faroit** [comment]
Just tried out the current implementation. For me the additional lid up notes are definitely an improvement. Have to try out ISOControllers layout again to check if this is any better.
